ConsoleApp1/Program.cs

```diff
@@ -2,8 +2,184 @@
 {
 	internal class Program
 	{
+		public enum SortAlgorithmType
+		{
+			SelectionSort,
+			BubbleSort,
+			InsertionSort
+		}
+
+		public enum OrderBy
+		{
+			Asc,
+			Desc
+		}
+
 		static void Main(string[] args)
 		{
+			Random random = new Random();
+			int length1 = random.Next(5,10);
+			int[] myArr = new int[length1];
+
+			for (int i = 0; i < length1 - 1; i++)
+			{
+				myArr[i] = random.Next(100);
+			}
+			Console.WriteLine("RANDOM VALUES ARRAY");
+			DisplayArray(myArr);
+
+			Console.WriteLine();
+			Console.WriteLine("Sorted random values array by Asc");
+
+			DisplayArray(Sort(ref myArr, SortAlgorithmType.InsertionSort, OrderBy.Asc));
+			Console.WriteLine();
+			Console.WriteLine();
+
+			Console.WriteLine("Sorted random values array by Desc");
+			DisplayArray(Sort(ref myArr, SortAlgorithmType.InsertionSort, OrderBy.Desc));
+			Console.WriteLine();
+			Console.WriteLine();
+
+		}
+
+		public static int[] SelectionSort(int[] array, OrderBy orderBy)
+		{
+			if (orderBy == OrderBy.Asc)
```

Comment:
You have almost identical 2 method's pieces, which differ only in comparing elements
So you need to 
`orderBy == OrderBy.Asc ? array[j] < array[minValue] : array[j] > array[minValue]`